cmd/zapi/cmd/post/pcap.go

```diff
@@ -0,0 +1,119 @@
+package post
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"sync/atomic"
+	"time"
+
+	"github.com/brimsec/zq/cmd/zapi/cmd"
+	"github.com/brimsec/zq/cmd/zapi/format"
+	"github.com/brimsec/zq/pkg/display"
+	"github.com/brimsec/zq/zqd/api"
+	"github.com/mccanne/charm"
+)
+
+var PcapPost = &charm.Spec{
+	Name:  "pcappost",
+	Usage: "pcappost [options] path",
+	Short: "post a pcap file to a space",
+	New:   NewPcap,
+}
+
+func init() {
+	cmd.CLI.Add(PcapPost)
+}
+
+type PcapCommand struct {
+	*cmd.Command
+	force      bool
+	bytesRead  int64
+	bytesTotal int64
+	done       bool
+}
+
+func NewPcap(parent charm.Command, flags *flag.FlagSet) (charm.Command, error) {
+	c := &PcapCommand{Command: parent.(*cmd.Command)}
+	flags.BoolVar(&c.force, "f", false, "create space if specified space does not exist")
+	return c, nil
+}
+
+// Run lists all spaces in the current zqd host or if a parameter
+// is provided (in glob style) lists the info about that space.
+func (c *PcapCommand) Run(args []string) (err error) {
+	if len(args) == 0 {
+		return errors.New("pcap path arg required")
+	}
+	var id api.SpaceID
+	client := c.Client()
+	if c.force {
+		sp, err := client.SpacePost(c.Context(), api.SpacePostRequest{Name: c.Spacename})
+		if err != nil && err != api.ErrSpaceExists {
+			return err
+		}
+		id = sp.ID
+	}
+	if id == "" {
+		id, err = c.SpaceID()
+		if err != nil {
+			return err
+		}
+	}
+	var dp *display.Display
+	if !c.NoFancy {
+		dp = display.New(c, time.Second)
+		go dp.Run()
+	}
+
+	file := args[0]
+	stream, err := client.PcapPost(c.Context(), id, api.PcapPostRequest{Path: file})
+	if err != nil {
+		return err
+	}
+	var res interface{}
+loop:
+	for {
+		res, err = stream.Next()
+		if err != nil {
+			break loop
+		}
+		if res == nil {
+			break loop
+		}
+		switch typ := res.(type) {
+		case *api.TaskEnd:
+			if typ.Error != nil {
+				err = typ.Error
+			}
+			break loop
+		case *api.PcapPostStatus:
+			atomic.StoreInt64(&c.bytesRead, typ.PcapReadSize)
+			atomic.StoreInt64(&c.bytesTotal, typ.PcapSize)
+		}
+	}
```

Comment:
```suggestion
	for {
		var v interface{}
		v, err = stream.Next()
		if err != nil || res == nil {
			break
		}
		switch v := v.(type) {
		case *api.TaskEnd:
			err = v.Error
			break
		case *api.PcapPostStatus:
			atomic.StoreInt64(&c.bytesRead, v.PcapReadSize)
			atomic.StoreInt64(&c.bytesTotal, v.PcapSize)
		}
	}
```

Reply:
The break in the switch won't exit the for loop.

Reply:
🤦